lib/nuget-parser/dotnet-core-parser.ts

```diff
@@ -45,61 +48,91 @@ function collectFlatList(targetObj) {
   });
 }
 
-function buildTreeRecursive(targetDeps, depName, parent, treeDepth) {
-  const MAX_TREE_DEPTH = 40;
-  if (treeDepth > MAX_TREE_DEPTH) {
-    throw new BigTreeError('The depth of the tree is too big.');
+function buildBfsTree(targetDeps, roots) {
+  let queue = [...roots];
+  const nodes: Dependency[] = [];
+  const links: DepLink[] = [];
+  while (queue.length > 0) {
+    const dep = queue.shift();
+    const foundPackage = findPackage(targetDeps, dep);
+    if (foundPackage && !isScanned(nodes, foundPackage)) {
+      nodes.push(foundPackage);
+      if (foundPackage.dependencies) {
+        addPackageDepLinks(links, foundPackage);
+        queue = queue.concat(Object.keys(foundPackage.dependencies));
+      }
+    }
   }
+  const tree = constructTree(roots, nodes, links);
+  return tree;
+}
+
+function isScanned(nodes: Dependency[], pkg: Dependency): boolean {
+  const node = nodes.find((elem) => elem.name === pkg.name && elem.version === pkg.version);
+  return !!node;
+}
 
-  let depResolvedName = '';
-  let originalDepKey = '';
+function isFreqDep(packageName: string): boolean {
+  return packageName in freqDeps;
```

Comment:
This syntax actually works?! TIL

Reply:
@orsagie this works :) but you need to be really careful :) cos in `freqDeps` is an object this will also return true: 
`'toString' in freqDeps`
`'__proto__' in freqDeps`
`'constructor' in freqDeps`  etc